solutions/task_2.py

```diff
@@ -0,0 +1,56 @@
+#!/usr/bin/env python3
+"""
+TASK: Create 'Fake' class to run code below without errors.
+"""
+
+import random
+import string
+
+
+class Fake(dict):
+    def __getattr__(self, name):
+        print("Non existing attribute {} has been called".format(name))
+        return self
+
+    def __call__(self, *args, **kwargs):
+        try:
+            dict.__call__(*args, **kwargs)
+            return self
+        except (TypeError, ValueError):
+            print("The attribute has been called with:")
+            if len(args) > 0:
+                print("- positional arguments: {}".format(str(", ".join(args))))
+            else:
+                print("- no positional arguments")
+            if len(kwargs) > 0:
+                kwargs_list = []
+                for k, v in kwargs.items():
+                    kwargs_list.append(str(k) + " = " + str(v))
+                print("- keyword arguments: {}".format(str(", ".join(kwargs_list))))
+            else:
+                print("- no keyword arguments")
+            return self
+
+    def __getitem__(self, item):
+        try:
+            return dict.__getitem__(self, item)
```

Comment:
**Important thing:**
In this task you have redefined behavior of some methods from parent class. -> So, for the first you a tying (`try:`) to behave in normal way (like parent object should behave). But you don't know how parent method should behave. Here you are truing to guess it 👎 

So if you are not Vanga you have to call this method from parent object. To do it right read about `super`.

Reply:
Python3: https://docs.python.org/3.5/library/functions.html#super
Python2: https://docs.python.org/2/library/functions.html#super
Python 2 notation is also supported with python 3.